jumanji/environments/packing/bin_pack/env.py

```diff
@@ -739,6 +746,214 @@ def _get_intersections_dict(
                 intersections_mask_dict[direction] &= ~to_remove
         return intersections_ems_dict, intersections_mask_dict
 
+    def merge_same_height_ems(self, state: State) -> None:
+        def compute_merge_mask(args: Tuple) -> Tuple:
+            """Function that takes an array of EMS and an array representing the EMS_mask and
+            returns a matrix of all the EMS that have the same height and can be merged to form a
+            bigger EMS.
+            Returns:
+                mask: 2D boolean array where mask[i,j] is True if tree_slice(EMS,i) and
+                    tree_slice(EMS, j) have the same height and can be merged.
+                same_x : 2D boolean array where same_x[i,j] is True if
+                    tree_slice(EMS,i).x1 == tree_slice(EMS,j).x1 and
+                    tree_slice(EMS,i).x2 == tree_slice(EMS,j).x2. This matrix is used to know along
+                    which axis of the container those two EMS should be merged if they're mergeable.
+            """
+
+            def isclose_matrix(a: chex.Array, b: chex.Array) -> chex.Array:
+                """
+                This function takes two 1D vectors and returns a 2D boolean matrix where
+                    (i,j) = True if a[i] is close to b[j].
+                Args:
+                    a (chex.Array):1D vector²
+                    b (chex.Array):1D vector²
+
+                Returns:
+                    chex.Array: _description_
+                """
+                return jnp.isclose(
+                    jnp.expand_dims(a, -1) - jnp.expand_dims(b, -1).transpose(),
+                    0,
+                )
+
+            ems_arr, ems_mask = args
+
+            same_y = isclose_matrix(ems_arr.y1, ems_arr.y1) & isclose_matrix(
+                ems_arr.y2, ems_arr.y2
+            )
+            same_x = isclose_matrix(ems_arr.x1, ems_arr.x1) & isclose_matrix(
+                ems_arr.x2, ems_arr.x2
+            )
+            side_by_side_x = isclose_matrix(ems_arr.x1, ems_arr.x2) | isclose_matrix(
+                ems_arr.x2, ems_arr.x1
+            )
+            side_by_side_y = isclose_matrix(ems_arr.y1, ems_arr.y2) | isclose_matrix(
+                ems_arr.y2, ems_arr.y1
+            )
+            # Two EMS can be merged if they have the same height and start at the same z1 and either
+            # have the same width and start at the same y1 and they're continuous on the x axis
+            # of the container,
```

Comment:
Could you fix the formatting of the comment?

Reply:
Could you also add a comment to explain that is_empty refers to an ems of volume zero (I know you didn't come up with the confusing terminology but it would help given that there's a lot going on in this line)

Reply:
done ✅